src/components/product/product.service.ts

```diff
@@ -146,6 +146,17 @@ export class ProductService {
     }
 
     if (input.produces) {
+      const produce = await this.db
+        .query()
+        .match([node('pr', 'Producible', { id: input.produces, active: true })])
+        .return('pr')
+        .first();
+      if (!produce) {
+        this.logger.warning(`Could not find producible node`, {
+          id: input.produces,
+        });
+        throw new NotFoundException('Could not find producible node');
```

Comment:
Can you change this `NotFoundException` import from nestjs/common to ours? This will allow you to pass the field that has the not found exception. Like this:
```suggestion
        throw new NotFoundException('Could not find producible node', 'product.produces');
```